File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n, int m, vector<int> section) {
    
    if(m == 1) return section.size();
    if(section.back()-section.front() < m) return 1;
    
    int NextPos = section[0]+m;
    int cnt = 1;
    for(int i = 1; i< section.size();i++)
    {
        if(section[i] < NextPos) continue;
        if(section[i] >= NextPos)
        {
            cnt++;
            NextPos = section[i]+m;
        }
        
    }
    return cnt;
}

File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.cs
using System;

public class Solution {
    public int solution(int n, int m, int[] section) {
        int answer = 0;
        if(m == 1) return section.Length;
        int clear = 0;
        for(int i = 0; i<section.Length;i++)
        {
            if(clear < section[i])
            {
                clear = section[i] + m -1;
                answer++;
            }
            else continue;
        }
        
        return answer;
    }
}